### ISSUE TITLE
close websockets at reset

### ISSUE THREAD

**Issue by Matthias-Hermsdorf** [espruino/EspruinoTools#83]
Hi,
the websocketserver does not die propper on reset() or reset(true).

Say I have this Server.
``
    function onPageRequest(req, res) {
       res.writeHead(200, {'Content-Type': 'text/html'});
        res.end("Hi. Ask the other Server");
    }

    var server = require('ws').createServer(onPageRequest);
    server.listen(80);
    console.log("server listen on port",80);

   server.on("websocket", function(ws) {
  	ws.on('message',function(msg) {
	    console.log("msg",msg);
        });
    });

I deploy the code with espruino -m -p tcp://192.168.178.56 socketTest.js to the Board, everything is fine.
I could connect to this SocketServer and can send Messages.

But if I want to deploy the next commit, I get a
    
    >ERROR: Socket bind failed
    --] Uncaught InternalError: Unable to create socket

This only happens if I establish a connection in between.
If the socketServer is unloaded, it should close ists connections.

I could store the sockets. But this variable isn't accessible at the next upload.

I use Espruino Command-line Tool 0.1.13 and Espruino 1.99

**Comment by Matthias-Hermsdorf** [comment]
Nice guess. Yes, it is a ESP32 and a 1.99

**Comment by gfwilliams** [comment]
Yeah, I'd ask on the forums as the folks on there know a lot more about it, but if you use a cutting edge build from http://www.espruino.com/binaries/travis/master/ you may find it just works :)

**Comment by Matthias-Hermsdorf** [comment]
My last build was five days old.
And with the current build it works. fine.

**Comment by gfwilliams** [comment]
It was fixed yesterday! https://github.com/espruino/Espruino/commit/5b07121317d0028a7253e3fa47c12b3383e09ceb
